src/Core/src/Application.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Maui
+{
+	public abstract class Application : IApplication
+	{
+		static object? GlobalLock;
+
+		IServiceProvider? _serviceProvider;
+		IMauiContext? _context;
+
+		protected Application()
+		{
+			GlobalLock = new object();
```

Comment:
This can't be set here as each ctor will create a new one. This probably should be in the static ctor.